client/src/main/java/com/google/cloud/pubsub/AbstractSubscriberConnection.java

```diff
@@ -310,7 +310,10 @@ public void run() {
 
   private void addOutstadingAckHandlers(
       ExpirationInfo expiration, final List<AckHandler> ackHandlers) {
-    outstandingAckHandlers.putIfAbsent(expiration, new ArrayList<AckHandler>(ackHandlers.size()));
```

Comment:
@denyska, @davidtorres all modification related to _outstandingAckHandlers_ state made in synchronized block by itself, means we are quite atomic here.

Please approve PR if you don't have other questions.